### PR TITLE
tests: add test for frame buffer interface

### PR DESCRIPTION
**Pull request by sergiocazzolato**
To test this use a device which has the file /dev/fb0, most of the cloud images don't have it.

### CODE REVIEW

File: tests/lib/snaps/test-snapd-framebuffer/bin/read [hunk 1]
```diff
@@ -0,0 +1,3 @@
+#!/bin/sh
+
+head -c 10 /dev/fb0
```
Comment (bboozzoo) [inline tests/lib/snaps/test-snapd-framebuffer/bin/read]: I'd probably make this a `dd if=/dev/fb0 of=/dev/null count=10`

File: tests/lib/snaps/test-snapd-framebuffer/bin/write [hunk 2]
```diff
@@ -0,0 +1,3 @@
+#!/bin/sh
+
+echo "$1" | dd bs=4 seek=$((100 * 1024 + 200)) >/dev/fb0
```
Comment (bboozzoo) [inline tests/lib/snaps/test-snapd-framebuffer/bin/write]: I think you mean:
`echo "$1" | dd bs=4 seek=$((100*1024 + 200)) of=/dev/fb0`

File: tests/main/interfaces-framebuffer/task.yaml [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+
+details: |
+    The framebuffer interface allows to access the /dev/fb* buffer files.
+
+    The test uses the test-snapd-framebuffer snap to write/read from /dev/fb0.
+
+    The test also checks the interface connection and disconnection works properly.
+
+prepare: |
+    snap try $TESTSLIB/snaps/test-snapd-framebuffer
```
Comment (stolowski) [inline tests/main/interfaces-framebuffer/task.yaml]: We should probably snap remove it in restore?

File: tests/lib/snaps/test-snapd-framebuffer/meta/snap.yaml [hunk 4]
```diff
@@ -0,0 +1,12 @@
+name: test-snapd-framebuffer
+version: 1.0
+summary: Basic snap to check access to file system
+description: A basic snap to check access to file system
```
Comment (stolowski) [inline tests/lib/snaps/test-snapd-framebuffer/meta/snap.yaml]: Both summary and description need changing, looks like a copy-paste from another test.

File: tests/main/interfaces-framebuffer/task.yaml [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+
+execute: |
+    CONNECTED_PATTERN=":framebuffer +test-snapd-framebuffer"
+    DISCONNECTED_PATTERN="\- +test-snapd-framebuffer:framebuffer"
+
+    if [ ! -e /dev/fb0 ]; then
+        echo "Framebuffer not available on /dev/fb0"
+        exit 0
```
Comment (stolowski) [inline tests/main/interfaces-framebuffer/task.yaml]: Is this intended, shouldn't this be an error? Or is it available on some of the systems only? Should we restrict this test to be run on select systems instead?

Comment (sergiocazzolato) [reply tests/main/interfaces-framebuffer/task.yaml]: That could be done for sure, but as I was asked to detect it dinamically the path in the uhid test (which is similat to this one), I though that should be better to make this detection dinamic here.

File: tests/main/interfaces-framebuffer/task.yaml (hunk 3 above)
Comment (sergiocazzolato) [reply tests/main/interfaces-framebuffer/task.yaml]: Thanks for reviewing this. Snaps are automatically removed in the reset.sh script which is called in the test setup.

File: tests/lib/snaps/test-snapd-framebuffer/meta/snap.yaml (hunk 4 above)
Comment (sergiocazzolato) [reply tests/lib/snaps/test-snapd-framebuffer/meta/snap.yaml]: done

File: tests/lib/snaps/test-snapd-framebuffer/bin/write (hunk 2 above)
Comment (sergiocazzolato) [reply tests/lib/snaps/test-snapd-framebuffer/bin/write]: done

**Pull request merged**